app/graphql/types/git/commit_type.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/BlockLength
+Types::Git::CommitType = GraphQL::ObjectType.define do
+  # rubocop:enable Metrics/BlockLength
+  name 'Commit'
+  description 'A commit of a repository'
+
+  field :id, !types.ID do
+    description 'The sha hash of the commit'
+  end
+
+  field :parentIds, !types[!types.ID] do
+    description 'The parents of the commit'
+    property :parent_ids
+  end
+
+  field :message, !types.String do
+    description 'The commit message'
+  end
+
+  field :author, !Types::Git::UserType do
+    description 'The author of the commit'
+
+    resolve(lambda do |commit, _arguments, _context|
+      OpenStruct.new(name: commit.author_name,
+                     email: commit.author_email,
+                     account: User.find(email: commit.author_email))
+    end)
+  end
+
+  field :authoredAt, !Types::TimeType do
+    description 'The time the commit was authored at'
+    property :authored_date
+  end
+
+  field :committer, !Types::Git::UserType do
+    description 'The committer of the commit'
+
+    resolve(lambda do |commit, _arguments, _context|
+      OpenStruct.new(name: commit.committer_name,
+                     email: commit.committer_email,
+                     account: User.find(email: commit.committer_email))
```

Comment:
Is it possible to make this call lazy? Otherwise the user will always be loaded, even if the account field is not requested.